### PR TITLE
Deployment preparation: `StakelessGaugeCheckpointer` (`L2GaugeCheckpointer` V2)

### PR DESCRIPTION
**Pull request by jubeira**
# Deployment preparation: `StakelessGaugeCheckpointer`

Build-info: https://github.com/balancer/balancer-v2-monorepo/commit/ea262dc756e99745c884383b95d3ed512ae3ef9d (master)

## Deployment preparation tasks checklist:

- [x] Deployments `README` is updated with the new task name and link
- [x] Task `README` has a description and links to the artifacts
- [x] `build-info` is updated with artifacts compiled at the specified commit
- [x] Artifacts are generated from `build-info`
- [x] New task has a working fork test 

Note: the task has two fork tests.
The base case should be identical to the one in the `20230527-l2-gauge-checkpointer` task, proving that the new version can still do the same.
The second one tests a corner case that was not covered by the original version.

## Deprecating old tasks checklist: 

- [x] Old task was moved to `deprecated/` directory
- [x] Deployments `README` is updated: old task is moved to the 'deprecated' section, and links are updated with the `deprecated/` prefix
- [x] Old task imports in `index.ts`, `input.ts` and fork tests are updated
- [x] Old task `README` is updated with a warning sign, a link to the replacement task and a short description 

## Code checklist:

- [x] The diff is legible and has no extraneous changes 
- [x] Complex code has been commented in tests and scripts
- [x] Fork test block number and test cases are appropriate for this deployment
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

N/A

### CODE REVIEW

File: tasks/20230731-stakeless-gauge-checkpointer/readme.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# 2023-05-31 - L2 Gauge Checkpointer
```
Comment (EndymionJkb) [inline tasks/20230731-stakeless-gauge-checkpointer/readme.md:1]:
```suggestion
# 2023-07-31 - Stakeless Gauge Checkpointer
```
Also describe the reason for the name change? (e.g., clarify that these are on L2)

**Comment by EndymionJkb** [comment]
I confirm that the build-info matches.

File: README.md [hunk 2]
```diff
@@ -118,13 +118,13 @@ Returns an object with all contracts from a deployment and their addresses.
...
 | Avalanche Root Gauge, for veBAL voting                 | [`20230529-avalanche-root-gauge-factory`](./tasks/20230529-avalanche-root-gauge-factory)             |
 | Timelock Authorizer, governance contract               | [`20230522-timelock-authorizer`](./tasks/20230522-timelock-authorizer)                               |
 | Pool Data Queries for bulk operations                  | [`20230613-balancer-pool-data-queries`](./tasks/20230613-balancer-pool-data-queries)                 |
 | Composable Stable Pools V5                             | [`20230711-composable-stable-pool-v5`](./tasks/20230711-composable-stable-pool-v5)                   |
 | L2 Child Chain Gauge Checkpointer (Relayer)            | [`20230712-child-chain-gauge-checkpointer`](./tasks/20230712-child-chain-gauge-checkpointer)         |
 | Chainlink Rate Provider Factory                        | [`20230717-chainlink-rate-provider-factory`](./tasks/20230717-chainlink-rate-provider-factory)       |
+| Stakeless Gauge Checkpointer                           | [`20230731-stakeless-gauge-checkpointer`](./tasks/20230725-stakeless-gauge-checkpointer)             |
```
Comment (EndymionJkb) [inline README.md]:
```suggestion
| Stakeless Gauge Checkpointer                           | [`20230731-stakeless-gauge-checkpointer`](./tasks/20230731-stakeless-gauge-checkpointer)             |
```

File: README.md [hunk 3]
```diff
@@ -195,3 +195,4 @@ Go to each deprecated deployment's readme file to learn more about why it is dep
 | Linear Pools for Gearbox Tokens                     | [`20230213-gearbox-linear-pool`](./tasks/deprecated/20230213-gearbox-linear-pool)                       |
 | Linear Pools for Silo Tokens                        | [`20230315-silo-linear-pool`](./tasks/deprecated/20230315-silo-linear-pool)                             |
 | Composable Stable Pools V4                          | [`20230320-composable-stable-pool-v4`](./tasks/deprecated/20230320-composable-stable-pool-v4)           |
+| L2 Gauge Checkpointer                               | [`20230527-l2-gauge-checkpointer`](./tasks/deprecated/20230527-l2-gauge-checkpointer)                   |
```
Comment (EndymionJkb) [inline README.md:198]: Need to run build-address-lookup to flip the status of the old l2-gauge-checkpointer to "Deprecated".

**Review by EndymionJkb — APPROVED** [review]
LGTM except for the comments noted that make CI pass

File: tasks/20230731-stakeless-gauge-checkpointer/readme.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# 2023-05-31 - Stakeless Gauge Checkpointer
```
Comment (EndymionJkb) [inline tasks/20230731-stakeless-gauge-checkpointer/readme.md:1]:
```suggestion
# 2023-07-31 - Stakeless Gauge Checkpointer
```
Sooo close :)

**Review by EndymionJkb — APPROVED** [review]
One last nit

File: tasks/20230731-stakeless-gauge-checkpointer/readme.md (hunk 4 above)
Comment (jubeira) [reply tasks/20230731-stakeless-gauge-checkpointer/readme.md]: ahhh dang, thanks!

**Review by EndymionJkb — APPROVED** [review]
Did it always do this "dismiss stale review" thing on every change, or did we just add that?

**Comment by jubeira** [comment]
> Did it always do this "dismiss stale review" thing on every change, or did we just add that?

It's a setting that was turned on for the deployments repo when it was set up (does not happen in the monorepo).

It's a bit annoying, but it also ensures that you cannot change the build info and go unnoticed. So I'm inclined to keeping it.
